packages/e2e-tests/src/util.js

```diff
@@ -0,0 +1,75 @@
+import execa from 'execa'
+import psTree from 'ps-tree'
+
+// let's try gracefully, otherwise we can do SIGTERM or SIGKILL
+const defaultKillSignal = 'SIGINT'
+const defaultLogger = process.env.DEBUG ? console.log : () => { }
+
+export async function startBackgroundProcess (
+  cmd,
+  readyOutput,
+  logger = defaultLogger,
+  killSignal = defaultKillSignal
+) {
+  return new Promise((resolve, reject) => {
+    // start the process
+    const subprocess = execa(cmd)
+    logger(cmd, 'spawned with PID: ', subprocess.pid)
+    let stdout = ''
+    let stderr = ''
+
+    // return this function so the process can be killed
+    const exit = () => new Promise((resolve, reject) => {
```

Comment:
The same issue we had with bash, doing `subprocess.kill()` would not stop a background process.

For example:
```
13388 ~ 'node: npm test'
  `- 9164 ~ 'node: ava'
    `- 9488 ~ 'node: execa'   <- this is what I get in the test, which does not want to die :(
      `- 11808 ~ 'node: aragon ipfs' 
        `- 24548 ~ 'ipfs' 
```

Even if we listen to the 'exit' event in the CLI:

```js
  process.on('exit', (code, signal) => {
    console.log('parent said exit', code, signal)
    ipfsProc.kill()
    process.exit()
  })
```

Most likely `child.pid` returns a Group ID, and node cannot kill groups on windows: https://github.com/nodejs/node/issues/3617

Solution: https://www.npmjs.com/package/ps-tree which returns the 2 children which we can kill. 🔪

Reply:
Great solution 🎉